experimental/uefi/runtime/src/echo.rs

```diff
@@ -0,0 +1,67 @@
+//
+// Copyright 2022 The Project Oak Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+extern crate alloc;
+
+use alloc::vec::Vec;
+use ciborium::{de, ser};
+
+#[derive(Debug)]
+pub enum Error<T> {
+    // An error occured while deserializing.
+    De(ciborium::de::Error<T>),
+
+    // An error occured while serializing.
+    Ser(ciborium::ser::Error<T>),
+}
+
+/// Basic hardware abstraction layer for sending data.
+pub trait Interface {
```

Comment:
nit: "Interface" is quite an overloaded term, so it might be worth considering a more specific name for clarity (perhaps Channel or CommsInterface?). Also, it seems that this trait is not specific to the echo , so I would consider moving it out of the echo module (either into lib.rs or its own module). Both of these could be addressed in future when non-echo functionality is added to the runtime.

Reply:
+1 for Interface being an overloaded term.